addons/bus/static/src/services/bus_service.js

```diff
@@ -60,21 +61,41 @@ export const busService = {
          * @param {{type: WorkerEvent, data: any}[]}  messageEv.data
          */
         function handleMessage(messageEv) {
-            const { type } = messageEv.data;
-            let { data } = messageEv.data;
-            if (type === "notification") {
-                data.forEach((d) => (d.message.id = d.id)); // put notification id in notif message
-                multiTab.setSharedValue("last_notification_id", data[data.length - 1].id);
-                data = data.map((notification) => notification.message);
-                for (const { id, type, payload } of data) {
-                    notificationBus.trigger(type, { id, payload });
+            const { type, data } = messageEv.data;
+            switch (type) {
+                case "notification": {
+                    const notifications = data.map(({ id, message }) => ({ id, ...message }));
+                    multiTab.setSharedValue("last_notification_id", notifications.at(-1).id);
+                    for (const { id, type, payload } of notifications) {
+                        notificationBus.trigger(type, { id, payload });
+                    }
+                    break;
+                }
+                case "initialized": {
+                    isInitialized = true;
+                    connectionInitializedDeferred.resolve();
+                    break;
+                }
+                case "outdated": {
+                    multiTab.unregister();
+                    notification.add(_t("The page appears to be out of date."), {
+                        title: _t("Refresh"),
+                        type: "warning",
+                        sticky: true,
+                        buttons: [
+                            {
+                                name: _t("Refresh"),
+                                primary: true,
+                                onClick: () => {
+                                    browser.location.reload();
+                                },
+                            },
+                        ],
+                    });
+                    break;
                 }
-            } else if (type === "initialized") {
-                isInitialized = true;
-                connectionInitializedDeferred.resolve();
-                return;
             }
-            if (type !== "notification") {
+            if (OUTGOING_EVENTS.has(type)) {
```

Comment:
This was previously a default, why the change (in stable)?

Reply:
Not clear what is emitted or not, some events are followed by "return" others are filtered with the condition here. Another type is added in this PR, which should not be emitted so I just explicitly listen what should be emitted. Nothing changes from an outside perspective, only events emitted are "connect", "disconnect", "reconnecting", "reconnect".

Reply:
But if for some reason somebody made a custo which relied on unsupported types, we're going to annoy them no?